### PR TITLE
Multiple cost

### PR DESCRIPTION
**Pull request by saranshbhardwaj**

### CODE REVIEW

File: RoadCare3/FormAnalysis.cs [hunk 1]
```diff
@@ -1341,5 +1403,201 @@ private void checkBoxMultipleCost_CheckedChanged(object sender, EventArgs e)
             }
 
         }
+
+        private void TabPagePriority_Click(object sender, EventArgs e)
```
Comment (ARA-Trans) [inline RoadCare3/FormAnalysis.cs]: Remove these empty functions.

File: RoadCare3/FormCalculatedField.cs [hunk 2]
```diff
@@ -208,25 +208,20 @@ private void MSSQLdgvCalculatedCellDoubleClick(DataGridViewCellEventArgs e)
 				if (dgvCalculated[nColumn, nRow].Value != null)
 				{
 					strEquation = dgvCalculated[nColumn, nRow].Value.ToString();
-                    isFunctionTag = dgvCalculated[0, nRow].Tag.ToString();
-                    if (isFunctionTag == "1") isFunction = true;
-				}
+                }
```
Comment (ARA-Trans) [inline RoadCare3/FormCalculatedField.cs]: Formatting

File: RoadCare3/FormLogin.cs [hunk 3]
```diff
@@ -377,7 +376,17 @@ private void FormLogin_Load(object sender, EventArgs e)
             chkUseIntegratedSecurity.Checked = Settings.Default.USE_INTEGRATED_SECURITY;
 
             tbRoadCareUserName.Text = Settings.Default.LAST_LOGIN;
-        
+            TabSqlServer.Enabled = true;
+            #if DEBUG
```
Comment (ARA-Trans) [inline RoadCare3/FormLogin.cs]: I don't think its a good idea to have this stuff in the form code. Someone can simply grab the database password from IL-Spy. This should be moved to a config file which would ideally be encrypted.

File: Simulation/BudgetCriteria.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                if (!File.Exists(Criteria.Evaluate.m_cr.PathToAssembly))
+                {
+                    Criteria.Evaluate = null;
+                }
+            }
+
+            Criteria.Criteria = criteria;
+            foreach (String str in Criteria.Errors)
+            {
+                SimulationMessaging.AddMessage(new SimulationMessage("Error: Supersede criteria for " + criteria + " :" + str));
```
Comment (ARA-Trans) [inline Simulation/BudgetCriteria.cs]: This line will need to change as we cannot use simulation messaging. Saransh are we just ignoring these lines?

File: Simulation/SimulationMessaging.cs [hunk 5]
```diff
@@ -77,6 +79,28 @@ public static class SimulationMessaging
         static private List<ConditionalRSL> _attributeConditionalRSL;
```
Comment (ARA-Trans) [inline Simulation/SimulationMessaging.cs]: This will likely need to be made [ThreadStatic]

File: Simulation/SimulationMessaging.cs [hunk 6]
```diff
@@ -77,6 +79,28 @@ public static class SimulationMessaging
...
         [ThreadStatic]
         static private Jurisdiction _jurisdiction;
+        [ThreadStatic]
+        static private List<RemainingLife> _attributeRemainingLife;
+        [ThreadStatic]
+        static private Dictionary<string,string> _noTreatmentRemainingLife;
+        [ThreadStatic]
+        static private List<BudgetCriteria> _budgetCriteria;
+
+        public static void ReleaseSimulationData()
```
Comment (ARA-Trans) [inline Simulation/SimulationMessaging.cs]: This will likely need to be made [ThreadStatic]

Comment (ARA-Trans) [reply Simulation/SimulationMessaging.cs]: All static members of the simulation messaging class need to be checked to see if they need to be ThreadStatic.

File: Simulation/BudgetCriteria.cs (hunk 4 above)
Comment (saranshbhardwaj) [reply Simulation/BudgetCriteria.cs]: We are ignoring these lines. Web application is not using it

**Pull request merged**